Library/Homebrew/formula.rb

```diff
@@ -405,11 +413,19 @@ def linked_keg
     Pathname.new("#{HOMEBREW_LIBRARY}/LinkedKegs/#{name}")
   end
 
+  def latest_head_prefix
+    head_prefixes = installed_prefixes.select do |pn|
+      pn.basename.to_s.start_with?("HEAD")
```

Comment:
I don't think you need to check `head_prefix_up_to_date` inside `eligible_kegs`. Noted, if `head_prefix_up_to_date` returns false, `installed?`(one line above `eligible_kegs =`) will return false as well.